core: add build patch for `sprintf` in macos builds

chenrui333 opened the pull request:
This pr fixes the following build issue on macos

```
/tmp/bareos-client-20231217-15168-qlx505/bareos-Release-23.0.0/core/src/tests/job_control_record.cc:62:5: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
    sprintf(jobs[i]->Job, "%d-test", 123 + i);
    ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
/tmp/bareos-client-20231217-15168-qlx505/bareos-Release-23.0.0/core/src/tests/job_control_record.cc:121:3: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
  sprintf(jcr->Job, "987-654-321");
  ^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
```

### Thank you for contributing to the Bareos Project!

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
~~Check backport line~~
- [x] Required backport PRs have been created

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

---

relates to https://github.com/Homebrew/homebrew-core/pull/157558

sebsura reviewed (changes requested):
Thanks for your contribution. The changes are great.  Please add yourself to the AUTHORS file, so we that this can be merged.

sebsura commented:
```
✗  Commit checks failed:
  bac6e7f30 chore: add build patch for `sprintf` in macos builds: body contains line longer 72 chars
✗  bareos-check-sources --since=1bbd14cf1d9c05071559fd3a98bd2e5dd6da6e9a reported:
  Plugin 'fix copyright year' would modify 'core/src/tests/job_control_record.cc'
```
pr-tool also reported these problems.

sebsura commented:
I fixed the problems and added you to the authors file.  If you did not want to be added, please leave a comment here!

chenrui333 commented:
> I fixed the problems and added you to the authors file. If you did not want to be added, please leave a comment here!

happy to be in the authors list, thanks!

chenrui333 commented:
Thanks @sebsura!
